lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -417,83 +216,25 @@ export abstract class ClientApplication {
         const loggedInAccounts = this.getAllAccounts();
         if (loggedInAccounts.length > 0) {
-            this.eventHandler.emitEvent(EventType.ACQUIRE_TOKEN_START, InteractionType.Popup, validRequest);
+            this.eventHandler.emitEvent(EventType.ACQUIRE_TOKEN_START, InteractionType.Popup, request);
         } else {
-            this.eventHandler.emitEvent(EventType.LOGIN_START, InteractionType.Popup, validRequest);
+            this.eventHandler.emitEvent(EventType.LOGIN_START, InteractionType.Popup, request);
         }
 
-        const browserRequestLogger = this.logger.clone(name, version, validRequest.correlationId);
-        const serverTelemetryManager = this.initializeServerTelemetryManager(ApiId.acquireTokenPopup, validRequest.correlationId);
-
-        try {
-            // Create auth code request and generate PKCE params
-            const authCodeRequest: CommonAuthorizationCodeRequest = await this.initializeAuthorizationCodeRequest(validRequest);
-
-            // Initialize the client
-            const authClient: AuthorizationCodeClient = await this.createAuthCodeClient(serverTelemetryManager, validRequest.authority, validRequest.correlationId);
-            browserRequestLogger.verbose("Auth code client created");
-
-            // Create acquire token url.
-            const navigateUrl = await authClient.getAuthCodeUrl(validRequest);
-
-            // Create popup interaction handler.
-            const interactionHandler = new PopupHandler(authClient, this.browserStorage, authCodeRequest, browserRequestLogger);
-
-            // Show the UI once the url has been created. Get the window handle for the popup.
-            const popupParameters: PopupParams = {
-                popup,
-                popupName
-            };
-            const popupWindow: Window = interactionHandler.initiateAuthRequest(navigateUrl, popupParameters);
-            this.eventHandler.emitEvent(EventType.POPUP_OPENED, InteractionType.Popup, {popupWindow}, null);
-
-            // Monitor the window for the hash. Return the string value and close the popup when the hash is received. Default timeout is 60 seconds.
-            const hash = await interactionHandler.monitorPopupForHash(popupWindow);
-            const state = this.validateAndExtractStateFromHash(hash, InteractionType.Popup, validRequest.correlationId);
-
-            // Remove throttle if it exists
-            ThrottlingUtils.removeThrottle(this.browserStorage, this.config.auth.clientId, authCodeRequest.authority, authCodeRequest.scopes);
-
-            // Handle response from hash string.
-            const result = await interactionHandler.handleCodeResponse(hash, state, authClient.authority, this.networkClient);
+        const popupClient = new PopupClient(this.config, this.browserStorage, this.browserCrypto, this.logger, this.eventHandler, this.navigationClient);
 
+        return popupClient.acquireToken(request).then((result) => {
```

Comment:
I'm not sure I see the value of mixing .then and async/await, but we can change this later.

Reply:
Can't do async/await here unfortunately. We made the popup APIs synchronous a while back to enable users to configure popups to be opened synchronously or asynchronously.